calc/CalcReader.scala

```diff
@@ -69,14 +89,27 @@ object CalcReader extends Reader:
                 addChild(Error("invalid number format", SourceMarker(m)))
                   *> rules
 
-  private lazy val lowPrecOpMode: Manip[SourceRange] =
-    commit:
-      consumeMatch: m =>
-        addChild(LowPrecOp(m))
-          *> rules
-
-  private lazy val highPrecOpMode: Manip[SourceRange] =
+  private lazy val operatorMode: Manip[SourceRange] =
     commit:
       consumeMatch: m =>
-        addChild(HighPrecOp(m))
-          *> rules
\ No newline at end of file
+        m.decodeString() match
+          case "+" =>
```

Comment:
This could be done directly on selecting above, using single char cases instead of oneOf then parsing which one it was.